Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -540,97 +563,112 @@ extension SourceKitServer: MessageHandler {
     // code completion session in sourcekitd but they don't need to be full
     // barriers to any other request.
     messageHandlingQueue.async(barrier: R.self is CompletionRequest.Type) {
-      let cancellationToken = CancellationToken()
-
-      let request = Request(
-        params,
-        id: id,
-        clientID: clientID,
-        cancellation: cancellationToken,
-        reply: { [weak self] result in
-          reply(result)
-          if let self {
-            Task {
-              await self._logResponse(result, id: id, method: R.method)
-            }
+      await withLoggingScope("request-\(id)") {
+        await self.handleImpl(params, id: id, from: clientID, reply: reply)
+      }
+
+    }
+  }
+
+  private func handleImpl<R: RequestType>(
+    _ params: R,
+    id: RequestID,
+    from clientID: ObjectIdentifier,
+    reply: @escaping (LSPResult<R.Response>) -> Void
+  ) async {
+    let startDate = Date()
+
+    let cancellationToken = CancellationToken()
+
+    let request = Request(
+      params,
+      id: id,
+      clientID: clientID,
+      cancellation: cancellationToken,
+      reply: { [weak self] result in
+        reply(result)
+        logger.log("Replied. Took \(-startDate.timeIntervalSinceNow * 1000)ms")
```

Comment:
Could we just pass in the time as part of `logResponse`?

Reply:
The problem is that more time could have elapsed until `_logResponse` gets executed (which is only on a background task because describing the response could take a while). If we wanted to log it in `_logReponse` we would still need to capture the date here, then pass the date to `_logResponse` and at that point, we might as well just log here.

Reply:
> If we wanted to log it in _logReponse we would still need to capture the date here

Yeah, that's what I meant.

> we might as well just log here

The downside as I see it is the extra log, when we already log the response 🤷. I'd just prefer to keep the amount of log spam down as much as we can.

Reply:
I inlined the `_log*` functions and unified the duration + response logging.

Reply:
Okay, that works too 🙇‍♂️. Thanks!